File: pset6/readability/readability.py
from cs50 import get_string

# Ask user for imput
text = get_string("Text: ")

# Counters
letter = 0
word = 1
sentence = 0

# Read string until it's end
for i in range(len(text)):

    # Check for lettersin text
    if 65 <= ord(text[i]) <= 90 or 97 <= ord(text[i]) <= 122:
        letter += 1

    # Check for end of word
    elif ord(text[i]) == 32:
        word += 1

    # Check for end of sentence
    elif ord(text[i]) == 33 or ord(text[i]) == 63 or ord(text[i]) == 46:
        sentence += 1

# Average words
L = (100 * letter) / word

# Average sentences
S = (100 * sentence) / word

# Calculate use algorithm
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print grade
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/Readability/readability.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //Variables for counting
    float letter = 0; float word = 1; float sentence = 0;
    
    //promp user for a text
    string achr = get_string("Text: ");
    
    //analize each ascii character
    for (int j = 0; achr [j] != 0; j++) // each charcacter of the string is analized by doing achr[j]
    {
        //letter count
        if ((65 <= achr[j] && achr[j] <= 90)|| (97 <= achr[j] && achr[j] <= 122)) //by conditioning it between numbers (intergers) the program deduces that achr must be regarded as it's numeric value
        {
            letter++;
        }
        //word count
        else if (achr[j] == 32 && achr[j+1] != 32)
        {
            word++;
        }
        //sentence count
        else if (achr[j] == 46 || achr[j] == 33 || achr[j] == 63) 
        {
            sentence++;
        }
    }
    //to ensure at least one sentence is counted
    if (sentence == 0)
        {
            sentence++;
        }
    //calculating the algorithm
    float L = (100 * letter) / word; //avarage letters for 100 words
    float S = (100 * sentence) / word; //avarage sentences for 100 words
    
    float grade = 0.0588 * L - 0.296 * S - 15.8;
        

    if (grade >= 16)  //if grade is 16 or higher
    {
        printf("Grade 16+\n");
    }
    else if (grade <= 1)  //if grade is 1 or lower
    {
        printf("Before Grade 1\n");
    }
    else  //if grade is beetwen 2 and 15
    {
    printf("Grade %.f\n", grade);
    }
}